File: algorithms/168.swift
// https://leetcode.com/problems/excel-sheet-column-title

class Solution {
    func convertToTitle(n: Int) -> String {
        let aScalars = "A".unicodeScalars
        let aCode = aScalars[aScalars.startIndex].value
        var n = n
        var result = [Int]()
        while n > 0 {
            n -= 1
            result.insert(n % 26, atIndex: 0)
            n /= 26
        }
        return String(result.map { Character(UnicodeScalar(aCode + UInt32($0))) })
    }
}

File: algorithms/168.py
# https://leetcode.com/problems/excel-sheet-column-title


class Solution(object):
    def convertToTitle(self, n):
        """
        :type n: int
        :rtype: str
        """
        ord_A = ord('A')
        result = []
        while n > 0:
            n, r = divmod(n - 1, 26)
            result.insert(0, r)
        return ''.join(chr(x + ord_A) for x in result)
